frappe/database.py

```diff
@@ -826,7 +826,7 @@ def flush_realtime_log(self):
 	def rollback(self):
 		"""`ROLLBACK` current transaction."""
 		self.sql("rollback")
-		self.begin()
+		#self.begin()
```

Comment:
why?

Reply:
I had the same question, why was it even there in the first place,  in a rollback of a transaction, that it is then starting a new one! - I was trying to do nested transactions for testing but it failed miserably, - Interestingly, it didn't change anything by removing this.  I can put it back in anyway, but it is not needed...